src/inspect_ai/model/_providers/anthropic.py

```diff
@@ -470,34 +474,41 @@ async def resolve_chat_input(
         # return chat input
         return system_param, tools_params, message_params, computer_use
 
+    ToolParamsResult = tuple[list["ToolParamDef"], bool]
+
     def tool_params_for_tools(
         self, tools: list[ToolInfo], config: GenerateConfig
-    ) -> tuple[list["ToolParamDef"], bool]:
-        # tool params and computer_use bit to return
-        tool_params: list["ToolParamDef"] = []
-        computer_use = False
-
-        # for each tool, check if it has a native computer use implementation and use that
-        # when available (noting that we need to set the computer use request header)
-        for tool in tools:
-            computer_use_tool = (
+    ) -> ToolParamsResult:
+        def reducer(
+            acc: AnthropicAPI.ToolParamsResult, tool: ToolInfo
+        ) -> AnthropicAPI.ToolParamsResult:
+            tool_param_def, computer_use = self.tool_param_for_tool_info(tool, config)
+            acc[0].append(tool_param_def)
+
+            return acc[0], acc[1] or computer_use
+
+        # for each tool, check if it has a native tool implementation and use that
+        # when available (noting names of the native tools used because we need to
+        # set the proper request header)
+        return functools.reduce(reducer, tools, (list[ToolParamDef](), False))
+
+    def tool_param_for_tool_info(
+        self, tool: ToolInfo, config: GenerateConfig
+    ) -> tuple["ToolParamDef", bool]:
+        if config.internal_tools is not False:
+            native_tool = (
                 self.computer_use_tool_param(tool)
-                if config.internal_tools is not False
-                else None
+                or self.text_editor_tool_param(tool)
+                or self.bash_tool_param(tool)
             )
-            if computer_use_tool:
-                tool_params.append(computer_use_tool)
-                computer_use = True
-            else:
-                tool_params.append(
-                    ToolParam(
-                        name=tool.name,
-                        description=tool.description,
-                        input_schema=tool.parameters.model_dump(exclude_none=True),
-                    )
-                )
+            if native_tool:
+                return native_tool, native_tool["name"] == "computer"
 
-        return tool_params, computer_use
+        return ToolParam(
+            name=tool.name,
+            description=tool.description,
+            input_schema=tool.parameters.model_dump(exclude_none=True),
+        ), False
```

Comment:
Since we now have to check for three different types of native Anthropic tools, I found it cleaner to convert this to a `reduce` that uses a helper function below to get the params for a particular `ToolInfo`